### PR TITLE
[hma] ThreatExchang/Collaboration config in HMAConfig, Helper Populate script

### PR DESCRIPTION
**Pull request by Dcallies**
Summary
---------

* Adds ThreatExchangeConfig
* Add a script to make it easy to populate test data (since old script doesn't work)

Doesn't actually pull the trigger on moving the rest over to give @mengyangwang  a chance to do more in #494 

Test Plan
---------

```
$  python -m hmalib.scripts.populate_config_db --help
usage: populate_config_db.py [-h] [--table TABLE] {update,load_example_configs,get} ...

Write simple configs to the HMA config database

positional arguments:
  {update,load_example_configs,get}
    update              Add or edit a config
    load_example_configs
                        Populate db with a default set of configs
    get                 get configs

optional arguments:
  -h, --help            show this help message and exit
  --table TABLE         The name of the config dynamodb table

$ python -m hmalib.scripts.populate_config_db update --help
usage: populate_config_db.py update [-h] {ThreatExchangeConfig} ...

positional arguments:
  {ThreatExchangeConfig}
    ThreatExchangeConfig
                        Update a config of this type

optional arguments:
  -h, --help            show this help message and exit

$ python -m hmalib.scripts.populate_config_db update ThreatExchangeConfig --help
usage: populate_config_db.py update ThreatExchangeConfig [-h] --enabled bool --privacy_group int str

positional arguments:
  str                  name

optional arguments:
  -h, --help           show this help message and exit
  --enabled bool       enabled
  --privacy_group int  privacy_group

$ python -m hmalib.scripts.populate_config_db update ThreatExchangeConfig Test --enabled false --privacy_group 1234
ThreatExchangeConfig(name='Test', enabled=False, privacy_group=1234)

$ python -m hmalib.scripts.populate_config_db load_example_configs
ThreatExchangeConfig(name='Test Config 1', enabled=False, privacy_group=303636684709969)
ThreatExchangeConfig(name='Test Config 2', enabled=False, privacy_group=258601789084078)

$ ThreatExchangeConfig(name='Test', enabled=False, privacy_group=1234)
ThreatExchangeConfig(name='Test Config 1', enabled=False, privacy_group=303636684709969)
ThreatExchangeConfig(name='Test Config 2', enabled=False, privacy_group=258601789084078)

$ python -m hmalib.scripts.populate_config_db get ThreatExchangeConfig --name Test
ThreatExchangeConfig(name='Test', enabled=False, privacy_group=1234)

$ python -m hmalib.scripts.populate_config_db get ThreatExchangeConfig --name Noexist
Traceback (most recent call last):
...
ValueError: No ThreatExchangeConfig named Noexist
```

### CODE REVIEW

**Review by eriksendc — APPROVED** [review]

File: hasher-matcher-actioner/hmalib/lambdas/fetcher.py [hunk 1]
```diff
@@ -55,33 +71,34 @@ def get(cls):
...
             s3_te_data_folder=os.environ["THREAT_EXCHANGE_DATA_FOLDER"],
-            collab_config_table=os.environ["THREAT_EXCHANGE_CONFIG_DYNAMODB"],
+            collab_config_table=os.environ["HMA_CONFIG_TABLE"],
         )
 
 
-def lambda_handler(event, context):
-    config = FetcherConfig.get()
+@dataclass
+class ThreatExchangeConfig(HMAConfig):
```
Comment (schatten) [inline hasher-matcher-actioner/hmalib/lambdas/fetcher.py:79]: Is `ThreatExchangeConfig` going to be used anywhere else?

Perhaps it is better located in a non-lambda-specific module?

File: hasher-matcher-actioner/hmalib/scripts/populate_config_db.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+from hmalib.common import config as hmaconfig
+
+SUPPORTED_CONFIGS = [
+    ThreatExchangeConfig,
+]
+
+
+def better_bool_type(s: str):
+    s = s.strip().lower()
+    if s in ("true", "1"):
```
Comment (schatten) [inline hasher-matcher-actioner/hmalib/scripts/populate_config_db.py:29]: Do you want to add "True" and "False" as well?

Comment (schatten) [reply hasher-matcher-actioner/hmalib/scripts/populate_config_db.py:29]: Since users are pythonic?

**Review by schatten — APPROVED** [review]
This definitely has functional and module overlap with #494. 

Approving to unblock. 

Notes:
* let's keep this PR only to adding the script.
* @mengyangwang will use this to rework #494 as he sees fit.

File: hasher-matcher-actioner/hmalib/lambdas/fetcher.py (hunk 1 above)
Comment (Dcallies) [reply hasher-matcher-actioner/hmalib/lambdas/fetcher.py:82]: Originally had it as its own module, but then ran into some issue (couldn't remember) that brought me back here. My plan is to refactor this whole module to slim it down in a followup

File: hasher-matcher-actioner/hmalib/scripts/populate_config_db.py (hunk 2 above)
Comment (Dcallies) [reply hasher-matcher-actioner/hmalib/scripts/populate_config_db.py:29]: lower() handles this

**Pull request merged**